types/convert_test.go

```diff
@@ -687,10 +687,26 @@ func (s *testTypeConvertSuite) TestConvert(c *C) {
 	signedAccept(c, mysql.TypeYear, 1901, "1901")
 	signedAccept(c, mysql.TypeYear, 1900.567, "1901")
 	signedDeny(c, mysql.TypeYear, 1900.456, "0")
+	signedAccept(c, mysql.TypeYear, 0, "0")
+	signedAccept(c, mysql.TypeYear, "0", "2000")
+	signedAccept(c, mysql.TypeYear, "00", "2000")
+	signedAccept(c, mysql.TypeYear, " 0", "2000")
+	signedAccept(c, mysql.TypeYear, " 00", "2000")
+	signedAccept(c, mysql.TypeYear, " 000", "0")
```

Comment:
I suppose that `"000"` will be converted to `"2000"`, does so `" 000"`?

Reply:
No. At least on the latest mariadb, it is not. My tests on mariadb showed that any string of length 4, started with a valid prefix char '0', is converted to "0" without adjust.

Reply:
Could you show some SQL test cases?

Reply:
```
create table t(a year);
insert ignore into t (a) values (" 0a");
insert into t (a) values (" 000");
insert ignore into t (a) values (" 0aa");
insert into t (a) values (" 00 ");
insert ignore into t (a) values (" 0a ");
select * from t;
```

The query result should be:

```
2000
0000
0000
0000
0000
```

Reply:
Thanks.